docs/ObserveApiProposal.md

```diff
@@ -0,0 +1,782 @@
+
+  END_WRITE
+
+  START_OBSERVE
+
+  START_READOUT
+
+  START_WRITE
+
+}
+
+type DatasetEvent {
+
+  <ExecutionEvent fields here, not repeated to focus on what is new>
+
+  eventType: DatasetStageType!
+
+  stepId: StepId!
+
+  datasetId: DatasetId!
+  
+  # Dataset, when available 
+  dataset: Dataset
+  
+}
+
+type Dataset {
+
+  datasetId: DatasetId!
+  
+  # File timestamp
+  timestamp: Timestamp!
+  
+  # DHS filename (probably made into a scalar limited by a regex)
+  filename: String!
+
+}
+
+```
+
+Multiple dataset events may refer to the same `stepId` because a single step may produce 
+multiple datasets.
+
+
+### `ExecutionConfiguration`
+
+The execution configuration details instrument-specific instructions for the current step and a
+guess at what future steps might contain.  This is where Observe obtains instrument configurations
+to apply as it executes a sequence. Because each instrument's features will differ, there must be
+specific instances for each instrument.
+
+```
+# Listing of available instruments.
+enum InstrumentType {
+  …
+  GMOS_N
+  GMOS_S
+  …
+}
+
+# Common across all instruments will be a field that returns the instrument 
+# type.
+interface ExecutionConfiguration {
+  instrument: InstrumentType!
+}
+
+type GmosNorthExecution implements ExecutionConfiguration {
+  …
+}
+
+type GmosSouthExecution implements ExecutionConfiguration {
+  …
+}
+
+```
+
+To do anything with the execution configuration (other than check the instrument type), the query
+will have to match on the specific configuration type.
+
+```
+query Execution($observationId: ObservationId!) {
+  execution(observationId: $observationId) {
+    exeuctionConfig {
+      instrument
+      ... on GmosNorthExecution {
+        static {
+          nodAndShuffle {
+            posA {
+              p { arcseconds }
+              q { arcseconds }
+            }
+            posB {
+              p { arcseconds }
+              q { arcseconds }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+```
+
+Most instrument configurations will likely have the same structure, albeit with different individual
+static and dynamic (step) types.  For example,
+
+```
+type GmosNorthExecution implements ExecutionConfiguration {
+
+  instrument: InstrumentType!
+
+  # Static configuration that cannot change over the course of the sequence
+  static:      GmosNorthStatic!
+  
+  # Acquisition sequence, if any, is requested until the user confirms that the
+  # target is acquired.  Some calibration observations, for example, will have
+  # no acquisition steps.
+  aquisition:  ExecutionSequence!
+  
+  # Science sequence describes how to configure the instrument to acheive its
+  # science goals. 
+  science:     ExecutionSequence!  
+  
+}
+
+# The execution sequence pairs the next atom that should be executed with a 
+# `possibleFuture` sequence for reference only.  Once the `nextAtom` is
+# complete, a new query is required to get the new `nextAtom`.
```

Comment:
Could an Atom that Observe already loaded be changed from Explore? What if Observe already started executing it?

Reply:
That's a good question.  It seems to me like we should prevent that in both the UI and API once execution has started.  The idea with atoms is that they be completed in their entirety or abandoned altogether.  Editing them while running seems anathema to that arrangement.